### PR TITLE
Improvement for 64bit Windows port

### PR DESCRIPTION
**Pull request by watsk**
Improvement for 64bit Windows port

Description
-----------

1. 64bit TickType_t is supported. (MSVC and MinGW)
The case `configTICK_TYPE_WIDTH_IN_BITS == TICK_TYPE_WIDTH_64_BITS` is added in portmacro.h.
It is straightforward extension from 16bit and 32bit.
This change is based on the following discussion on the FreeRTOS forum.
https://forums.freertos.org/t/64bit-application-on-freertos-windows-port/19110

2. Unnecessary compiler warning is disabled locally. (MinGW-w64 only)
#pragma for ignoring the warning `cast-function-type` is added in port.c.
MinGW-w64 reports this warning for the for the cast operation from `TaskFunction_t` to `LPTHREAD_START_ROUTINE`.
This warning should be ignored because it is unavoidable. `TaskFunction_t` is an essential type in FreeRTOS and `LPTHREAD_START_ROUTINE` is an essential type in win32api.
This pragma option affects only one line. Other part is not affected.

Test Steps
-----------

Define `configTICK_TYPE_WIDTH_IN_BITS` as `TICK_TYPE_WIDTH_64_BITS` in FreeRTOSConfig.h and build FreeRTOS by MinGW-w64 or  x64 platform on MSVC(Visual Studio).

Regression tests are performed.
All combinations of tick type width and compiler type have been tested.
1. TICK_TYPE_WIDTH_32_BITS on MinGW(32bit) and Win32 on MSVC
2. TICK_TYPE_WIDTH_64_BITS on MinGW(32bit) and Win32 on MSVC
3. TICK_TYPE_WIDTH_32_BITS on MinGW-w64 and x64 on MSVC
4. TICK_TYPE_WIDTH_64_BITS on MinGW-w64 and x64 on MSVC

MingW Demo: CodeCoverage configuration runs successfully in all cases. No error is reported in the console. No regression is found on coverage score.
MSVC Demo: Demo application runs successfully. No error is reported in the console.

Checklist:
----------

- [x] I have tested my changes. No regression in existing tests.
- [ ] I have modified and/or added unit-tests to cover the code changes in this Pull Request.

Related Issue
-----------

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by chinglee-iot** [comment]
@watsk 
Thank you for creating this PR. I will look into it and discuss with you.

**Comment by watsk** [comment]
@chinglee-iot
Thank you for reviewing this PR. If you have any question, please let me know.

**Review by chinglee-iot — APPROVED** [review]
Thank you for creating this PR. The information is very clear to me.

File: portable/MSVC-MingW/port.c [hunk 1]
```diff
@@ -246,8 +246,19 @@ StackType_t * pxPortInitialiseStack( StackType_t * pxTopOfStack,
                                                FALSE,  /* Start not signalled. */
                                                NULL ); /* No name. */
 
+
+#ifdef 	__x86_64__
```
Comment (aggarg) [inline portable/MSVC-MingW/port.c:250]: Why is this needed only on x64 and not on Win32?

Comment (watsk) [reply portable/MSVC-MingW/port.c]: @aggarg
Thank you for pointing out. You are correct. Win32 is also needed.

I added `#pragma` only for x64 because I observed that MinGW32 does not report warning and I thought it is x64 unique.
But I found that MinGW32 version on my PC is too old and the latest MinGW32 reports same warning as MinGW64.
I modified `__x86_64__` to` __GNUC__`.
(I didn't delete `#ifdef` because MSVC does not recognize this pragma directive.)

Could you review modification?
I tested again all combinations of `TickType_t` width and compiler width on MinGW and MSVC.
No problem was found.

Comment (aggarg) [reply portable/MSVC-MingW/port.c]: Thank you for your contribution. Looks good!

**Review by aggarg — APPROVED** [review]

**Review by chinglee-iot — APPROVED** [review]
